### PR TITLE
[ACS-3219] Delete Rule

### PR DESCRIPTION
**Pull request by nikita-web-ua**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [X] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

There is currently no way for a user to delete a rule set for a folder.

**What is the new behaviour?**

The user can now delete the selected rule. 

Screenshot 2022-08-25 at 14 23 27

Clicking the "delete" button will open a confirmation dialog.

Screenshot 2022-08-25 at 14 23 55

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/aca-folder-rules/src/lib/rule-details/rule-details.ui-component.ts [hunk 1]
```diff
@@ -72,6 +72,8 @@ export class RuleDetailsUiComponent implements OnInit, OnDestroy {
       this._initialValue = newValue;
     }
   }
+  @Input()
+  onRuleDelete: () => void;
```
[the remark this answers is not in the document]
Comment (nikita-web-ua) [reply projects/aca-folder-rules/src/lib/rule-details/rule-details.ui-component.ts:76]: My fault, I forgot to delete.

**Review by TomikaArome — APPROVED** [review]

File: projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.scss [hunk 2]
```diff
@@ -21,6 +21,37 @@
     padding: 32px;
     overflow: scroll;
 
+    &__preview{
```
Comment (DenysVuika) [inline projects/aca-folder-rules/src/lib/manage-rules/manage-rules.smart-component.scss:24]: slightly wrong formatting, missing spaces before brackets, and the color of the "p" should be coming from the CSS variable to support theming

**Review by DenysVuika — CHANGES_REQUESTED** [review]

**Comment by montgolfiere** [comment]
@nikita-web-ua 

In summary: can you please confirm when the "delete" option appears for the selected rule ?

In more detail (Tl;Dr;):

I believe the current list of ordered rules can include

1. rules inherited from parent folders
2a. either: rules linked from another folder (rule set)
2b. or: rules directly defined on the current folder

This new functionality makes sense in terms of 2b. 

Can you please confirm what happens in the case of 1 and 2a ? 

In other words, do we show and allow the delete (at least for now) if the selected rule inherited or linked (as long as the current user has appropriate permission on that folder) ? 

If yes, do we have a Jira to go back and re-work in the future, eg. to allow the user to first navigate to the parent/linked folder (and then decide whether to delete the rule or not, if permissions allow) ?

**Comment by nikita-web-ua** [comment]
> I believe the current list of ordered rules can include:
> 
> * (1) rules inherited from parent folders (if any)
> * (2a) either: rules linked from another folder (rule set)
> * (2b) or: rules directly defined on the current folder

Yes, currently list includes rules from all scenarios you mentioned above. But works as supposed only in (2b).  In other scenarios delete button is still shown, but it will result to nothing as backend throws "500" error.

**Comment by montgolfiere** [comment]
Thanks. 500 doesn't sound good (:bug:) :wink:

It sounds like we need two jiras
- fix BE 500 (either return suitable 4xx if valid error case or else 2xx if valid success case)
- improve FE UI for dealing with inherited / linked rules - ie. no delete icon but rather a way to navigate to the appropriate folder

**Review by DenysVuika — APPROVED** [review]
